tests/js/server/aql/aql-read-own-writes.js

```diff
@@ -0,0 +1,170 @@
+/*jshint globalstrict:false, strict:false, sub: true, maxlen: 500 */
+/*global assertEqual, assertFalse, assertTrue, AQL_EXPLAIN */
+
+////////////////////////////////////////////////////////////////////////////////
+/// @brief tests for read own writes
+///
+/// DISCLAIMER
+///
+/// Copyright 2010-2012 triagens GmbH, Cologne, Germany
+///
+/// Licensed under the Apache License, Version 2.0 (the "License");
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+///     http://www.apache.org/licenses/LICENSE-2.0
+///
+/// Unless required by applicable law or agreed to in writing, software
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+///
+/// Copyright holder is triAGENS GmbH, Cologne, Germany
+///
+/// @author Jan Steemann
+/// @author Copyright 2012, triAGENS GmbH, Cologne, Germany
+////////////////////////////////////////////////////////////////////////////////
+
+const db = require("@arangodb").db;
+const jsunity = require("jsunity");
+
+function readOwnWritesSuite () {
+  const cn = "UnitTestsCollection";
+      
+  const getForLoops = function (nodes) {
+    return nodes.filter((n) => n.type === 'EnumerateCollectionNode' || n.type === 'IndexNode');
+  };
+
+  return {
+
+    setUp : function () {
+      db._drop(cn);
+      db._create(cn);
+    },
+
+    tearDown : function () {
+      db._drop(cn);
+    },
+    
+    testNonCollectionForLoop : function () {
+      const query = `FOR i IN 1..3 RETURN i`;
+      let nodes = AQL_EXPLAIN(query).plan.nodes;
```

Comment:
could these nodes variables be const because you initialize them, but don't edit them later on?

Reply:
Yes, they could be const. But it won't make a difference here.

Reply:
Changed anyway.